bittensor/neuron.py

```diff
@@ -91,6 +91,7 @@ def subscribe(self, synapse: bittensor.Synapse):
         logger.info("subscribe synapse: {}", synapse_proto)
         self._metagraph.subscribe(synapse_proto)
         self._axon.subscribe(synapse_proto, synapse)
+        synapse.setup_optimizer()
```

Comment:
This looks good although we will need to document this well. 
Another way to do this would be making the optimizer a property of the synapse. I think that might be cleaner.